### PR TITLE
migration 2.2

### LINKED ISSUE #156: compatibility issue with 2.2
PIM-6367: Rename service pim_enrich.query.product_and_product_model_query_builder_factory into pim_catalog.query.product_and_product_model_query_builder_factory

```
    pim_custom_entity.event_subscriber.check_reference_data_on_removal:
        class: '%pim_custom_entity.event_subscriber.check_reference_data_on_removal.class%'
        arguments:
            - '@pim_custom_entity.repository.attribute'
            - '@pim_catalog.query.product_and_product_model_query_builder_factory'
            - '@pim_custom_entity.configuration.registry'
            - '@doctrine.orm.entity_manager'
        tags:
            - { name: kernel.event_subscriber }
```

### PR DESCRIPTION
**Pull request by mmetayer**
Fixes #156

### CODE REVIEW

**Comment by jmleroux** [comment]
I updated the description to have the link between PR and issue

**Comment by mmetayer** [comment]
Actually, this is PR is still not complete (as it is only compatible with PIM >= 2.2).

File: Tests/AbstractTestCase.php [hunk 1]
```diff
@@ -5,7 +5,10 @@
 use Akeneo\Bundle\BatchBundle\Command\BatchCommand;
 use Akeneo\Test\IntegrationTestsBundle\Configuration\CatalogInterface;
 use Akeneo\Test\IntegrationTestsBundle\Security\SystemUserAuthenticator;
+use Doctrine\ORM\EntityManagerInterface;
```
Comment (jmleroux) [inline Tests/AbstractTestCase.php]: Why do we need new `use` clauses without changes in the class?

Comment (fitn) [reply Tests/AbstractTestCase.php]: Because it is used in phpdoc ;)

**Pull request merged**